src/Stubs/EloquentBuilder.stubphp

```diff
@@ -353,12 +372,662 @@ class Builder
 
     /**
      * @param TModel $model
-     * @return self<TModel>
```

Comment:
I remember I had to do this hack a while back -- glad to see the core issue in psalm has been resolved!